SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -0,0 +1,806 @@
+    }
+
+    switch (self.transportSelection) {
+        case SDLTransportSelectionTCP:
+            return [self sdl_startTCPSecondaryTransport];
+        case SDLTransportSelectionIAP:
+            return [self sdl_startIAPSecondaryTransport];
+        default:
+            SDLLogW(@"Unknown transport type for secondary transport: %ld", (long)self.transportSelection);
+            return NO;
+    }
+}
+
+- (BOOL)sdl_disconnectSecondaryTransport {
+    if (self.secondaryTransport == nil) {
+        SDLLogD(@"Secondary transport is already stopped.");
+        return NO;
+    }
+
+    SDLLogD(@"Disconnect secondary transport");
+    [self.secondaryTransport disconnect];
+    self.secondaryTransport = nil;
+
+    return YES;
+}
+
+- (BOOL)sdl_startTCPSecondaryTransport {
+    if (![self sdl_isTCPReady]) {
+        SDLLogD(@"Cannot start TCP secondary transport");
+        return NO;
+    }
+
+    SDLLogD(@"Starting TCP Secondary Transport (IP address = \"%@\", port = \"%d\")", self.ipAddress, self.tcpPort);
+
+    SDLTCPTransport *transport = [[SDLTCPTransport alloc] init];
+    transport.hostName = self.ipAddress;
+    transport.portNumber = [NSString stringWithFormat:@"%d", self.tcpPort];
+    SDLProtocol *protocol = [[SDLProtocol alloc] init];
+    transport.delegate = protocol;
+    protocol.transport = transport;
+    [protocol.protocolDelegateTable addObject:self];
+
+    self.secondaryProtocol = protocol;
+    self.secondaryTransport = transport;
+
+    // we reuse Session ID acquired from primary transport's protocol
+    // this is for Register Secondary Transport frame
+    [self.secondaryProtocol storeHeader:self.primaryProtocolListener.primaryRPCHeader forServiceType:SDLServiceTypeControl];
+    // this is for video and audio services
+    [self.secondaryProtocol storeHeader:self.primaryProtocolListener.primaryRPCHeader forServiceType:SDLServiceTypeRPC];
+
+    [self.secondaryTransport connect];
+    return YES;
+}
+
+- (BOOL)sdl_startIAPSecondaryTransport {
+    SDLLogD(@"Starting Secondary Transport: iAP");
+
+    SDLIAPTransport *transport = [[SDLIAPTransport alloc] init];
+    SDLProtocol *protocol = [[SDLProtocol alloc] init];
+    transport.delegate = protocol;
+    protocol.transport = transport;
+    [protocol.protocolDelegateTable addObject:self];
+
+    self.secondaryProtocol = protocol;
+    self.secondaryTransport = transport;
+
+    // we reuse Session ID acquired from primary transport's protocol
+    // this is for Register Secondary Transport frame
+    [self.secondaryProtocol storeHeader:self.primaryProtocolListener.primaryRPCHeader forServiceType:SDLServiceTypeControl];
+    // this is for video and audio services
+    [self.secondaryProtocol storeHeader:self.primaryProtocolListener.primaryRPCHeader forServiceType:SDLServiceTypeRPC];
+
+    [self.secondaryTransport connect];
+    return YES;
+}
+
+- (BOOL)sdl_isTCPReady {
+    if ([self.ipAddress length] == 0) {
+        SDLLogD(@"IP address is empty");
```

Comment:
No logs are needed in this method

Reply:
Hmm, `sdl_isTCPReady` can return `NO` for various reasons. So I thought debug logs would be helpful for developers. I can remove them, but I prefer to keep.